src/topology/metric_space/basic.lean

```diff
@@ -865,6 +865,22 @@ instance metric_space_pi : metric_space (Πb, π b) :=
     { refine eq.symm (comp_sup_eq_sup_comp _ _ _),
       exact (assume x y h, ennreal.coe_le_coe.2 h), refl },
     simp [dist, edist_nndist, ennreal.of_real, A]
+  end,
+  to_uniform_space := Pi.uniform_space _,
+  uniformity_dist := begin
+    -- with simp only on next line, the proof fails for no reason...
+    simp [Pi.uniformity, uniformity_dist, comap_infi, comap_infi, gt_iff_lt, preimage_set_of_eq,
+          comap_principal],
+    rw infi_comm, congr, funext ε,
+    rw infi_comm, congr, funext εpos,
+    simp only [ext_iff, εpos, dist, principal_eq_iff_eq, prod.forall, mem_Inter, mem_set_of_eq,
+               infi_principal_fintype],
+    assume a b,
+    let ε' : nnreal := ⟨ε, le_of_lt εpos⟩,
+    have A : ε' = nnreal.of_real ε, by simp [nnreal.of_real, ε', le_of_lt εpos],
+    change (∀ (i : β), dist (a i) (b i) < ε) ↔ (sup univ (λ (i : β), nndist (a i) (b i))) < ε',
+    simp only [finset.sup_lt_iff (show ⊥ < ε', from εpos)],
+    simp [nndist_dist, A, εpos],
```

Comment:
It doesn't block this PR, but this proof appears redundant with the emetric proof in the other file. There should be a way to construct a metric space from an emetric space and a dist function that equals the edist (and you shouldn't have to prove anything else about the dist, because it all follows from properties of the edist).

Reply:
There was already a way to construct a metric space from an emetric space, but one could not choose a formula for the distance which was not defeq to the one coming from the edistance. I have modified slightly this constructor to allow for a choice of `dist`, and used it for the construction of the pi instance.